File: scala/gigasecond/src/main/scala/Gigasecond.scala
import java.time.LocalDate
import java.time.LocalTime
import java.time.LocalDateTime

object Gigasecond {
  private final val GIGASECOND = 1000000000
  def addGigaseconds(input: LocalDate): LocalDateTime = {
    LocalDateTime.of(input, LocalTime.of(0, 0)).plusSeconds(GIGASECOND)
  }
  def addGigaseconds(input: LocalDateTime): LocalDateTime = {
    input.plusSeconds(GIGASECOND)
  }

}

File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

class Gigasecond {
    public final static long GIGASECOND = 1_000_000_000;
    LocalDateTime birthDateTime;
    Gigasecond(LocalDate birthDate) {
        this.birthDateTime = LocalDateTime.of(birthDate.getYear(), birthDate.getMonth(), birthDate.getDayOfMonth(), 0, 0);
    }

    Gigasecond(LocalDateTime birthDateTime) {
        this.birthDateTime = birthDateTime;
    }

    LocalDateTime getDateTime() {
        return this.birthDateTime.plusSeconds(GIGASECOND);
    }


}
